### ISSUE TITLE
You don't need separate `regex` and `iregex` combinators.

### ISSUE THREAD

**Issue by ekmett** [ChrisPenner/lens-regex-pcre#2]
You don't need separate `regex` and `iregex` combinators.

```haskell
regex :: IndexedTraversal Int 
regex = conjoined regex' (indexing regex')
```

where `regex'` is the existing `regex` recovers all of the performance of `regex` when the index is not in use.

This even opens up some cute options like making your quasiquoter (or another one) just return the traversal rather than the regex object. Not sure that is worth it, but it'd reduce syntactic noise even further.

```haskell
>>> "one _two_ three _four_" ^.. [rx|_\w+_|] . match
```

is rather pretty.

**Comment by ChrisPenner** [comment]
Ahh, very nice! I knew there was a way to do this, but wasn't sure what it looked like. I'll throw this in the next version ASAP 👍 

I've thought about inline quasi-quoter a little (I had a similar idea); but for now I think I'll leave it as is (at least until things settle down a little). It's pretty, but I think it might hurt 'hoogle-ability' too much.

There's the additional problem that people will often want to pass `PCREOption`s to their Regex's; the current structure allows them to pre-compile their regexes with all the options they require and pass them in to `regex` as named expressions which I think improves readability.

E.g.

```haskell
lineNumbersPat :: Regex
lineNumbersPat = compile [multiline] "^\d+"

srcText ^.. regex lineNumbersPat . match
```

Which encourages readability and re-use. We could also offer a `makeRegexTraversalQQ :: [PCREOption] -> QuasiQuoter` which does the best of both worlds; I'll keep it in mind as things evolve. 😄

**Comment by ChrisPenner** [comment]
Done in 5af285dfeb655b6ebb735701ee9b93d54b5d4251; available in `0.3.0.0`
